lib/compat/wordpress-6.7/blocks.php

```diff
@@ -43,3 +43,58 @@ function gutenberg_filter_block_type_metadata_settings_allow_variations_php_file
 	return $settings;
 }
 add_filter( 'block_type_metadata_settings', 'gutenberg_filter_block_type_metadata_settings_allow_variations_php_file', 10, 2 );
+
+/**
+ * Filter the query vars (tax_query) for the Query Loop block to support post formats.
+ *
+ * @param array    $query The query vars.
+ * @param WP_Block $block Block instance.
+ * @return array   The filtered query vars.
+ */
+function gutenberg_filter_query_loop_block_query_vars_post_format( $query, $block ) {
+	if ( isset( $block->context['query']['format'] ) &&
+		! empty( $block->context['query']['format'] ) ) {
+
+		$formats   = $block->context['query']['format'];
+		$tax_query = array( 'relation' => 'OR' );
+
+		// The default post format, 'standard', is not saved in the database.
+		// If 'standard' is part of the request, the query needs to exclude the formats.
+		if ( in_array( 'standard', $formats, true ) ) {
+			$tax_query[] = array(
+				'taxonomy' => 'post_format',
+				'field'    => 'slug',
+				'terms'    => array(),
+				'operator' => 'NOT EXISTS',
+			);
+			// Remove the standard format:
+			$formats = array_diff( $formats, array( 'standard' ) );
```

Comment:
it would probably be fine to simply `unset( $formats['standard'] )` here instead of calling the `array_diff()` function, given that we know the key we want to remove.

Reply:
`standard` is a value so unset alone does not work.

```suggestion
			unset( $formats[ array_search( 'standard', $formats, true ) ] );
```